packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1500,7 +1500,8 @@ class _CarouselPosition extends ScrollPositionWithSingleContext implements _Caro
     } else {
       item = getItemFromPixels(oldPixels, oldViewportDimensions ?? viewportDimension);
     }
-    final double newPixels = getPixelsFromItem(item, flexWeights, itemExtent);
+    final double newPixels =
+        (viewportDimension == 0.0) ? 0.0 : getPixelsFromItem(item, flexWeights, itemExtent);
```

Comment:
Can we move this check in `getPixelsFromItem` so that we can remove the change in line 1408 as well? Just think it would be better if we move the logic into one method.

Reply:
Great call! PR updated 🙏